### ISSUE TITLE
Config.php requires "use Aws\\Credentials\\CredentialProvider"

### ISSUE THREAD

**Issue by andrea-malatesta-ws** [aws-samples/aws-refarch-moodle#31]
Hi,
I have an infrastructure with moodle installed as described in aws-refarcg-moodle.

Since August 4th I have a 500 HTTP server error and I found the following error.

#######################################################################
Fatal errorPHP Fatal error:  Uncaught Aws\\Exception\\CredentialsException: Error retrieving credentials from the instance profile metadata service. (cURL error 6: Could not resolve host: https (see https://curl.haxx.se/libcurl/c/libcurl-errors.html)) in phar:///var/www/moodle/html/lib/aws.phar/Aws/Credentials/InstanceProfileProvider.php:272
Stack trace:
#0 phar:///var/www/moodle/html/lib/aws.phar/Aws/Credentials/InstanceProfileProvider.php(159): Aws\\Credentials\\InstanceProfileProvider->handleRetryableException(Object(GuzzleHttp\\Exception\\ConnectException), Array, 'Error retrievin...')
#1 [internal function]: Aws\\Credentials\\InstanceProfileProvider->Aws\\Credentials\\{closure}()
#2 phar:///var/www/moodle/html/lib/aws.phar/GuzzleHttp/Promise/Coroutine.php(160): Generator->throw(Object(GuzzleHttp\\Exception\\ConnectException))
#3 phar:///var/www/moodle/html/lib/aws.phar/GuzzleHttp/Promise/Promise.php(204): GuzzleHttp\\Promise\\Coroutine->_handleFailure(Object(GuzzleHttp\\Exception\\ConnectException))
#4 phar:///var/www/moodle/html/lib/aws.phar/Guzzl in phar:///var/www/moodle/html/lib/aws.phar/Aws/Credentials/InstanceProfileProvider.php on line 272

#######################################################################

To resolve it I have to change the initial part of config.php file from:

                  require 'lib/aws.phar';
                  use Aws\\SecretsManager\\SecretsManagerClient;
                  use Aws\\Exception\\AwsException;
                  use Aws\\Iam\\IamClient;

                  $client = new SecretsManagerClient([
                      'version' => '2017-10-17',
                      'region' => ''
                  ]);

to:

                  require 'lib/aws.phar';
                  use Aws\\SecretsManager\\SecretsManagerClient;
                  use Aws\\Exception\\AwsException;
                  use Aws\\Iam\\IamClient;
                  use Aws\\Credentials\\CredentialProvider;

                  $provider = CredentialProvider::defaultProvider();

                  $client = new SecretsManagerClient([
                      'version' => '2017-10-17',
                      'region' => '',
                      'credentials' => $provider
                  ]);

Basically seems that a breaking change has been introduced in aws.phar.

Please can you help me to understand  how to prevent that from happening? 
Is it available an url point for a specific v3 revision ?
What I did is correct ?
You too have the same anomaly as me?

Thanks

**Comment by brianjbeach** [comment]
Can you confirm that you have an IAM profile assigned to the instance and the metadata service is enabled?  It should look something like this.

```
% aws ec2 describe-instances
{
    "Reservations": [
        {
            "Groups": [],
            "Instances": [
                {
                    ...
                    "IamInstanceProfile": {
                        "Arn": "arn:aws:iam::XXXXXXXXXXXX:instance-profile/XXXXXX",
                        "Id": "XXXXXXXXXXXXX"
                    },
                    ...
                    "MetadataOptions": {
                        "State": "applied",
                        "HttpTokens": "optional",
                        "HttpPutResponseHopLimit": 1,
                        "HttpEndpoint": "enabled"
                    },
                    ...
                }
            ],
            ...
        }
    ]
}
```

**Comment by andrea-malatesta-ws** [comment]
Yes. I can confirm  it.
IAM profile is assigned to instance and the metadata service is enabled.

**Comment by vrioux** [comment]
Is this issue still valid? If so, please reopen. Thanks!
